rest_framework/settings.py

```diff
@@ -19,9 +19,15 @@
 back to the defaults.
 """
 from django.conf import settings
-from django.test.signals import setting_changed
 from django.utils.module_loading import import_string
 
+# Import `setting_changed` from the core package instead of its official
+# location if possible, to avoid importing the test module unnecessarily.
+try:
+    from django.core.signals import setting_changed
+except ImportError:
+    from django.test.signals import setting_changed
```

Comment:
The signal import path [is documented](https://docs.djangoproject.com/en/4.1/ref/signals/#setting-changed) as `django.test.signals.setting_changed`. So technically the `django.core.signals` is an implementation detail that is not guaranteed to be stable. So I added the fallback only for future compat. That said, I don't foresee it changing, and even if it does change, it can be fixed in a new release.

Should I remove the fallback?

Reply:
🤷🏼‍♂️ Seems low risk to me.

Reply:
It's gone!